### PR TITLE
chore: fix invalid mount in 'make start'

### PR DESCRIPTION
**Pull request by alexmt**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -54,7 +54,7 @@ define run-in-test-server
 		-v ${GOCACHE}:/tmp/go-build-cache${VOLUME_MOUNT} \
 		-v ${HOME}/.kube:/home/user/.kube${VOLUME_MOUNT} \
 		-v /tmp:/tmp${VOLUME_MOUNT} \
-		-v /app/config:/app/config${VOLUME_MOUNT} \
+		-v /tmp/argo-e2e/app/config:/app/config${VOLUME_MOUNT} \
```
Comment (jannfis) [inline Makefile:57]: Hm, I think this is not the correct fix. `/tmp/argo-e2e` only exists during E2E tests, and lives within the container. This mount is for running ArgoCD locally, and it should persist the application configuration (TLS certs, SSH known hosts and also GPG keys in the future) somewhere. But I agree, `/app/config` might not be the correct place on the host.

**Review by jannfis — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (jannfis) [reply Makefile:57]: After thinking about it for a while, I think we can remove the mount all together - if `/app/config` needs updating within the container, it should be done using `docker cp` - the data does not need to be persistent.

File: Makefile [hunk 2]
```diff
@@ -9,7 +9,7 @@ GIT_COMMIT=$(shell git rev-parse HEAD)
 GIT_TAG=$(shell if [ -z "`git status --porcelain`" ]; then git describe --exact-match --tags HEAD 2>/dev/null; fi)
 GIT_TREE_STATE=$(shell if [ -z "`git status --porcelain`" ]; then echo "clean" ; else echo "dirty"; fi)
 PACKR_CMD=$(shell if [ "`which packr`" ]; then echo "packr"; else echo "go run vendor/github.com/gobuffalo/packr/packr/main.go"; fi)
-VOLUME_MOUNT=$(shell if test selinuxenabled; then echo ":Z"; elif test "$(go env GOOS)"=="darwin"; then echo ":delegated"; else echo ""; fi)
```
Comment (alexmt) [inline Makefile:12]: It appears that `test selinuxenabled`  returns true on mac. Switched checks to ensure that `$(go env GOOS)"=="darwin"` verified first

File: Makefile [hunk 3]
```diff
@@ -373,7 +372,7 @@ clean: clean-debug
 .PHONY: start
 start:
 	docker version
-	$(call run-in-test-server,make ARGOCD_PROCFILE=test/container/Procfile start-local)
+	$(call run-in-test-server,make ARGOCD_PROCFILE=test/container/Procfile start-local ARGOCD_START=${ARGOCD_START})
```
Comment (alexmt) [inline Makefile:375]: Added ability to run components independently. E.g. `make start ARGOCD_START=api-server`

File: test/fixture/testrepos/start-git.sh [hunk 4]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
 docker run --name e2e-git --rm -i \
-    -p 2222:2222 -p 9080:80 -p 9443:443 -p 9444:444 -p 9445:445 \
```
Comment (alexmt) [inline test/fixture/testrepos/start-git.sh:4]: Bug fix.

File: Makefile (hunk 2 above)
Comment (jannfis) [reply Makefile:12]: :+1:

File: test/fixture/testrepos/start-git.sh (hunk 4 above)
Comment (jannfis) [reply test/fixture/testrepos/start-git.sh:4]: :+1:

**Review by jannfis — APPROVED** [review]
LGTM, thank you for fixing!

**Pull request merged**